test/test_nestedtensor.py

```diff
@@ -1923,6 +1929,34 @@ def test_empty_like(self, device, dtype):
         with self.assertRaisesRegex(RuntimeError, "empty_like only supports contiguous memory format for Nested Tensors"):
             nt_empty = torch.empty_like(nt_noncont)
 
+    def _assert_real_fake_equivalent(self, real_nt, fake_nt):
```

Comment:
Do you think that we should also check the sizes, strides , offsets..  via the private functions instead of just unbinding. I know in theory equivalent but IMO I think these are the attributes that define NT v NT equivalence could be convinced otherwise

Reply:
Well, we probably should, but the metadata functions are currently not exposed (even as private functions) and I'm trying hard to avoid doing so in this PR. This may be a foolish mission, but I think it's worthwhile to put off the exposure as long as possible, and we may find in the long term we don't need to at all. This would be nice because then there's no possibility of someone depending on it.

That said, I'm all for ensuring testing is complete; what are we doing elsewhere in testing to compare nested tensors?

Reply:
Yeah if you look in test_nestedtensor.py 
```
        nt_sizes = nt._nested_tensor_size()
        nt_strides = nt._nested_tensor_strides()
```
exists.  Still hidden since we don't want to encourage users to use this size info in its tensor format

Reply:
Hm fair enough - I wonder if it's worth limiting their usages so we can eventually remove them?

I see that `is_same_size` is defined for nested - can we just use that?

Reply:
Although these functions are marked temporary, we might need to access the sizes, strides and offsets of the FakeTensor in inductor :(

Reply:
> Hm fair enough - I wonder if it's worth limiting their usages so we can eventually remove them?
> 
> I see that `is_same_size` is defined for nested - can we just use that? or introduce something broader that also compares `strides` / `offsets`?

I actually wrote that bad boy like 9 months ago before really had non-contiguous NTs so it only checks that nt_size == nt_size not the strides or offsets

Reply:
@eellison Can I just chuck the tests currently in this PR in favor of using `CrossRefFakeMode`-based tests once that supports NT?

Reply:
Sounds good